### PR TITLE
Adds Tests for CAS Service Validate

### PR DESCRIPTION
**Pull request by StetsenTech**
**Ticket(s)**:
https://app.asana.com/0/1133543009734854/1150084536382593/f

**Summary**:
- Adds test for /serviceValidate
- Cleans up and converts logic into Rails format

### CODE REVIEW

File: app/controllers/cas_controller.rb [hunk 1]
```diff
@@ -301,30 +301,32 @@ def validate
...
     @renew = params['renew']
 
+    c = request.env['HTTP_X_FORWARDED_FOR'] || request.env['REMOTE_HOST'] || request.env['REMOTE_ADDR']
+
     st, @error = validate_service_ticket(@service, @ticket)
-    @success = st && !@error
+
+    if @error
+      return render json: {error: @error}, status: :unprocessable_entity
+    end
```
Comment (hcli-work) [inline app/controllers/cas_controller.rb]: `return render(json: {error: @error}, status: :unprocessable_entity) if @error`

File: app/controllers/cas_controller.rb [hunk 2]
```diff
@@ -301,30 +301,32 @@ def validate
...
+    c = request.env['HTTP_X_FORWARDED_FOR'] || request.env['REMOTE_HOST'] || request.env['REMOTE_ADDR']
+
     st, @error = validate_service_ticket(@service, @ticket)
-    @success = st && !@error
+
+    if @error
+      return render json: {error: @error}, status: :unprocessable_entity
+    end 
+
+    @success = !st.nil?
```
Comment (hcli-work) [inline app/controllers/cas_controller.rb]: Is there a chance that `st` can be the `false` object? If not, then a more idiomatic way to write this is
```
@success = !!st
```

File: app/controllers/cas_controller.rb (hunk 1 above)
Comment (StetsenTech) [reply app/controllers/cas_controller.rb]: Oh snap! I know this syntax, but didn't think of using it for this!

File: app/controllers/cas_controller.rb (hunk 2 above)
Comment (StetsenTech) [reply app/controllers/cas_controller.rb]: Found what I have seen, it's either an object or `nil`.

File: app/controllers/cas_controller.rb [hunk 3]
```diff
@@ -301,30 +301,32 @@ def validate
...
 
     if @success
       @username = st.username
       if @pgt_url
         pgt = generate_proxy_granting_ticket(@pgt_url, st, c)
         @pgtiou = pgt.iou if pgt
       end
-      @extra_attributes = st.granted_by_tgt.extra_attributes || {}
+      tgt = TGT.find_by({id: st.ticket_granting_ticket_id})
```
Comment (hcli-work) [inline app/controllers/cas_controller.rb]: you can leave the `{}` off

**Pull request merged**